src/libprojectM/wipemalloc.cpp

```diff
@@ -25,21 +25,61 @@
  */
 
 #include "wipemalloc.h"
+#include <assert.h>
 
- void *wipemalloc( size_t count ) {
-    count = (count + 15) & ~(size_t)15;
-    void *mem = aligned_alloc( 16, count );
+ void *wipemalloc( size_t count )
+ {
+    void *mem = malloc( count );
     if ( mem != NULL ) {
         memset( mem, 0, count );
       } else {
         printf( "wipemalloc() failed to allocate %d bytes\n", (int)count );
       }
     return mem;
-  }
+ }
 
 /** Safe memory deallocator */
  void wipefree( void *ptr ) {
     if ( ptr != NULL ) {
         free( ptr );
       }
   }
+
+void *wipe_aligned_alloc( size_t align, size_t size )
+{
+#if TARGET_OS_MAC
+    // only support powers of 2 for align
+    assert( (align & (align-1)) == 0 );
+    void *allocated = malloc(size + align - 1 + sizeof(void*));
+    if (allocated == NULL)
+    {
+        printf( "wipe_aligned_malloc() failed to allocate %d bytes\n", (int)size );
+        return NULL;
+    }
+    void *ret = (void*) (((size_t)allocated + sizeof(void*) + align -1) & ~(align-1));
+    *((void**)((size_t)ret - sizeof(void*))) = allocated;
+    return ret;
+#else
+    void *mem = aligned_alloc( align, size );
```

Comment:
is aligned_alloc available on EVERY other platform that isn't apple? raspi? BSD? windows?
I know mac is special and needs ^2 alignment, that's fine. i'm concerned about the aligned_alloc being portable.

Reply:
I honestly don't know.  It is C11 standard, and the only google references I found about lack of support seem related to OSX.  I could remove the #ifdef and always use the hand-written version.

Reply:
Well do whatever you think is best

Reply:
MacOS seems to contain posix_memalign, would that not be sufficient? It seems to work (almost) exactly the same except takes an address to write the pointer to.

Reply:
There's this from stackoverflow